include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2359,7 +2568,8 @@ class binary_reader
             }
 
             // reverse byte order prior to conversion if necessary
-            if (is_little_endian != InputIsLittleEndian)
+            if ((is_little_endian != InputIsLittleEndian && format != input_format_t::bjdata) ||
+                    (is_little_endian == InputIsLittleEndian && format == input_format_t::bjdata))
```

Comment:
thanks, maybe you had spotted an error here - my goal is to define a `is_swapbyte` flag, which is triggered
- if the system endian is LE, and input binary JSON format is ubjson/msgpack/cbor, or
- if the system endian is BE and the input binary format is bjdata

the old json.hpp does not have `InputIsLittleEndian` and this logic is simpler, now I am a little confused about when is `InputIsLittleEndian` supposed to be used. if this flag indicates input is always LE, then, perhaps I should have used this?
```
if(format == input_format_t::bjdata)
    InputIsLittleEndian = true;  // overwriting InputIsLittleEndian because bjdata is always LE
if ((is_little_endian != InputIsLittleEndian)
{
   ...
}
```

Reply:
I'm not sure what it should be, I was just trying to interpret your conditional.

Reply:
now I see why `InputIsLittleEndian` was needed - BSON is also little-endian by default, so it was used for handle BSON data

I now simplified the condition to 
```
if (is_little_endian != (InputIsLittleEndian || format == input_format_t::bjdata))
```
but it could be further simplified to 
```
if (is_little_endian != (format == input_format_t::bson || format == input_format_t::bjdata))
```
and remove `InputIsLittleEndian` from the parameter list, but I will leave this decision for the developers.